Built fine for me on x86_64-darwin, but I did get a few test failures. My mac is old and clunky, so it could be that.

```
builder for '/nix/store/ia2ns7izrbkdw3s75cxhykgaqk7dgx57-igraph-0.9.3.drv' failed with exit code 2; last 10 log lines:
         369 - test::scg3 (Failed)
      381 - test::igraph_adjacency_spectral_embedding (Failed)
  Errors while running CTest
  make[3]: *** [tests/CMakeFiles/check.dir/build.make:77: tests/CMakeFiles/check] Error 8
  make[3]: Leaving directory '/private/tmp/nix-build-igraph-0.9.3.drv-0/source/build'
  make[2]: *** [CMakeFiles/Makefile2:6072: tests/CMakeFiles/check.dir/all] Error 2
  make[2]: Leaving directory '/private/tmp/nix-build-igraph-0.9.3.drv-0/source/build'
  make[1]: *** [CMakeFiles/Makefile2:6079: tests/CMakeFiles/check.dir/rule] Error 2
  make[1]: Leaving directory '/private/tmp/nix-build-igraph-0.9.3.drv-0/source/build'
  make: *** [Makefile:2751: check] Error 2
```

--- a/pkgs/development/libraries/igraph/default.nix
+++ b/pkgs/development/libraries/igraph/default.nix
@@ -77,15 +77,17 @@ stdenv.mkDerivation rec {
     "-DIGRAPH_USE_INTERNAL_GMP=OFF"
     "-DIGRAPH_GLPK_SUPPORT=ON"
     "-DIGRAPH_GRAPHML_SUPPORT=ON"
-    "-DIGRAPH_ENABLE_LTO=ON"
+    "-DIGRAPH_ENABLE_LTO=AUTO"
     "-DIGRAPH_ENABLE_TLS=ON"
     "-DBUILD_SHARED_LIBS=ON"
   ];
 
   doCheck = true;
 
-  preCheck = ''
-    # needed to find libigraph.so
+  # needed to find libigraph, and liblas on darwin
+  preCheck = if stdenv.isDarwin then ''
+    export DYLD_LIBRARY_PATH="${lib.makeLibraryPath [ blas ]}:$PWD/src"
+  '' else ''
     export LD_LIBRARY_PATH="$PWD/src"
   '';
 

